tools/benchmark-campaign/runner.py

```
@@ -0,0 +1,2139 @@
+        raise ContractError(
+            f"pair {pair_index} must contain one candidate and one Core arm"
+        )
+    candidate = by_role[Role.CANDIDATE]
+    core = by_role[Role.CORE]
+    valid = candidate.run_valid and core.run_valid
+    candidate_result = candidate.arm_result
+    core_result = core.arm_result
+    correctness: bool | None = None
+    if (
+        candidate_result is not None
+        and not candidate_result.correctness_ok
```

Ruff (0.16.1)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)